UofT-Frosh-Orientation/orientation-website
# 600 sponsor layout change

## **Contribution:**

### **Issue:**
Closes #600 

### **Accomplishments:**
- Added 'View Less' button that sends you back into carousel view for sponsors by editing Home.jsx -> HomePageSponsors. Added another button and set corresponding `setViewAll` variable to false on click.
- Made all sponsor images in the carousel have the same size by editing ImageCarousel.jsx and ImageCarousel.scss - there was some inline styling in the `carousel-slide-container` using the 'transform' property that made each image different sizes
- Added mesh gradient borders for each sponsor in carousel view (only middle one's border is shown)
   - Created another field in sponsors.jsx for the 'rank', used this for inline CSS styling for each item in the carousel
   - Added each mesh gradient (bunch of radial-gradients combined) into colors.jsx

### **Visuals:** 
'View Less' button: 
![image](https://github.com/UofT-Frosh-Orientation/orientation-website/assets/109243682/9fc05f83-7d56-457c-b6be-9a64e259903b)

Image Carousel (Diamond, Gold, Bronze):
![image](https://github.com/UofT-Frosh-Orientation/orientation-website/assets/109243682/98bee9d0-9f78-4712-9ed6-c87c4e41b178)
![image](https://github.com/UofT-Frosh-Orientation/orientation-website/assets/109243682/c62b21bd-4fcd-4ef0-9dce-bbe65d0df236)
![image](https://github.com/UofT-Frosh-Orientation/orientation-website/assets/109243682/76c6310d-ddd7-46a2-9dbb-4b11f8d3d33e)

## **Details:**

### **Expected Behavior:**
- 'View Less' button should collapse 'all sponsors' viewing and send user back to carousel view
- Sponsor images should be of same size and have visual indication of the rank (bronze, diamond, gold, etc.) from their border

### **Testing Steps:**

Steps required to get expected behavior:
- Just run the front end and scroll down on the homepage to the sponsor carousel

### **Complications:**
- Mesh gradients finally worked in the borders somewhat, not sure if it works in mobile view though since I had some laptop issues as well
- I wasn't sure how to include all the radial-gradients in separate lines in the colors.jsx file, so they're just these huge lines right now: 
![image](https://github.com/UofT-Frosh-Orientation/orientation-website/assets/109243682/dc8d0ab6-d82a-4eec-899d-ecd8c3662a13)
- I also made the 'no rank' sponsors just have a white border, but if there's a preferred colour for that, I can add it! Let me know if other colours should be changed as well

### **Resources:**

Any additional resources used outside this in issue.

- Linear gradient maker: [https://cssgradient.io/](url)
- Mesh gradient maker (courtesy of Natalie): [https://csshero.org/mesher/](url)
